lib/class/json_data.class.php

```diff
@@ -0,0 +1,638 @@
+     * nothing fancy here...
+     *
+     * @param    integer    $code    Error code
+     * @param    string    $string    Error message
+     * @return    string    return error message JSON
+     */
+    public static function error($code,$string)
+    {
+        $JSON = json_encode(array("error" => array("code" => $code, "message" => $string)), JSON_PRETTY_PRINT);
+        return $JSON;
+    } // error
+
+    /**
+     * single_string
+     *
+     * This takes two values, first the key second the string
+     *
+     * @param    string    $key    (description here...)
+     * @param    string    $string    JSON data
+     * @return    string    return JSON
+     */
+    public static function single_string($key, $string='')
+    {
+        if (!empty($string)) {
+            $JSON = json_encode(array($key => $string), JSON_PRETTY_PRINT);
+        } else {
+            $JSON = json_encode(array("message" => $key), JSON_PRETTY_PRINT);
+        }
+
+        return $JSON;
+    } // single_string
+
+    /**
+     * tags_string
+     *
+     * This returns the formatted 'tags' string for an JSON document
+     *
+     */
+    private static function tags_string($tags)
+    {
+        $JSON = null;
+
+        if (is_array($tags)) {
+            $atags = array();
+            foreach ($tags as $tag_id => $data) {
+                if (array_key_exists($data['id'], $atags)) {
+                    $atags[$data['id']]['count']++;
+                } else {
+                    $atags[$data['id']] = array('name' => $data['name'],
+                        'count' => 1);
+                }
+            }
+
+            foreach ($atags as $id => $data) {
+                $JSON['id'] = $id;
+                $JSON['count'] = $data['count'];
+                $JSON['name'] = $data['name'];
+            }
+        }
+
+        return $JSON;
+    } // tags_string
+
+
+    /**
+     * playlist_song_tracks_string
+     *
+     * This returns the formatted 'playlistTrack' string for an JSON document
+     *
+     */
+    private static function playlist_song_tracks_string($song, $playlist_data)
+    {
+        if (empty($playlist_data)) {
+            return "";
+        }
+        $playlist_track = "";
+
+        foreach ($playlist_data as $playlist) {
+            if ($playlist["object_id"] == $song->id) {
+                return $playlist["track"];
+            }
+        }
+        return "";
+    } // playlist_song_tracks_string
+
+    /**
+     * tags
+     *
+     * This returns tags to the user, in a pretty JSON document with the information
+     *
+     * @param    array    $tags    (description here...)
+     * @return    string    return json
+     */
+    public static function tags($tags)
+    {
+        if (count($tags) > self::$limit or self::$offset > 0) {
+            $tags = array_splice($tags,self::$offset,self::$limit);
+        }
+
+        $JSON = [];
+
+        foreach ($tags as $tag_id) {
+            $tag    = new Tag($tag_id);
```

Comment:
Not sure about the policy for aligning variables declaration in Ampache. Sometimes it is aligned, sometimes it is not. Would need @Afterster input on this.